src/stores/index.js

```diff
@@ -267,13 +313,21 @@ const Store = Reflux.createStore({
    * validate instead the existing connection object.
    */
   async onConnectClicked() {
-    this.StatusActions.showIndeterminateProgressBar();
+    if (this.state.isConnecting) {
+      return;
+    }
+
+    this.setState({
+      isConnecting: true
+    });
+    const connectionAttemptId = uuidv4();
```

Comment:
isn't this only useful if more than one connection attempt is active at the same time?

Reply:
Yes - I think I was getting a little ahead of myself with that implementation. I'll update it and add a nice class like the one in schema analysis.